core/src/main/scala/com/rallyhealth/vapors/factfilter/dsl/Dsl.scala

```diff
@@ -9,14 +9,12 @@ import com.rallyhealth.vapors.factfilter.data._
 
 private[dsl] class Dsl extends TypedFactOps {
 
-  // TODO: Use some cats typeclass instead of iterable?
-  def all[F[x] <: IterableOnce[x], T, A](cond: CondExp[T]): CondExp[F[T]] = liftCondExp {
-    ExpAlg.ForAll[F[T], T, Boolean](identity[F[T]], cond, True, False)
+  def forall[T, V](cond: CondExp[V])(implicit ev: T <:< IterableOnce[V]): CondExp[T] = liftCondExp {
```

Comment:
This is because I want to use the early breaking "forall" and "exists" methods on the collections, (rather than computing all of the expressions in parallel and then ignoring the ones that are not needed to prove the query true). I'm not sure if `IterableOnce` will be the right solution, but it is a common supertype of `Option` and `Iterable`. I mainly only use this for facts that have nested data structures, so there isn't much concern about these not being in memory. Maybe someday I want to allow a fact to have a stream of data... but I can't see that for the forseeable future.

Long story short, this is safe for now, since I only traverse it once, but if I need more capabilities in the future, it shouldn't be hard to refactor because this is only used for data that has facts with non-scalar data fields.